src/compositor/output.c

```diff
@@ -874,6 +874,47 @@ wlc_output_set_resolution(wlc_handle output, const struct wlc_size *resolution)
    wlc_output_set_resolution_ptr(convert_from_wlc_handle(output, "output"), resolution);
 }
 
+bool
+wlc_output_set_scale_ptr(struct wlc_output *output, int32_t scale)
+{
+   if (!output)
+      return false;
+
+   assert(scale);
+   output->information.scale=scale;
+   return true;
+}
+
+WLC_API void
+wlc_output_set_scale(wlc_handle output, int32_t scale)
+{
+   wlc_output_set_scale_ptr(convert_from_wlc_handle(output, "output"), scale);
+}
+
+WLC_API int32_t
+wlc_output_get_scale(wlc_handle output)
+{
+   struct wlc_output *o = convert_from_wlc_handle(output, "output");
+   assert(o);
+   return o->information.scale;
+}
+
+WLC_API bool
+wlc_output_get_scaled_size(wlc_handle output, struct wlc_size *out_size)
+{
+   if (!out_size)
```

Comment:
assert instead, as out_size is not optional parameter at all to this. Though I'm questioning the usefulness of this function. It also might make sense to zero initialize out_size. It would make possible to use this function then regardless of return value as long as zero size is fine.

But for now, I think this function can be removed as I don't think of it as neccessity, writing wrapper function should be straight forward.